packages/editor/src/plugins/text/hooks/use-suggestions.tsx

```diff
@@ -91,147 +113,205 @@ export const useSuggestions = (args: useSuggestionsArgs) => {
-    })
-  }
+  })
 
   function insertSelectedPlugin(pluginType: EditorPluginType | string) {
     editor.deleteBackward('line')
 
-    // If the text plugin is selected from the suggestions list, clear the editor
     if (pluginType === EditorPluginType.Text) {
-      // In browsers other than chrome, the cursor is sometimes in front of the `/`
       editor.deleteForward('line')
+      setShowSuggestions(false)
+      refocus(editor)
       return
     }
 
     setTimeout(() => {
-      // Split the text-plugin and insert selected new plugin
       insertPlugin({ pluginType, editor, id, dispatch })
 
-      // If there is an empty line on the initial selection point, remove it
       if (selection && editor.children[selection.anchor.path[0]]) {
         editor.deleteBackward('block')
       }
     })
-  }
-
-  function handleSuggestionsMenuClose(event: KeyboardEvent) {
-    if (!closure.current.showSuggestions) return
-
-    event.preventDefault()
-
-    editor.deleteBackward('line')
+    setShowSuggestions(false)
   }
 
   return {
     showSuggestions,
+    setShowSuggestions,
     suggestionsProps: {
-      options,
-      suggestionsRef,
-      selected,
-      onMouseDown: insertSelectedPlugin,
-      onMouseMove: setSelected,
+      basicOptions,
+      interactiveOptions,
+      itemRefs,
+      searchInputRef,
+      focusSearchInput,
+      searchString,
+      setSearchString,
+      currentlyFocusedItem,
+      setCurrentlyFocusedItem,
+      insertSelectedPlugin,
     },
   }
 }
 
+interface PluginStrings {
+  title?: string
+  description?: string
+}
+
 function createOption(
-  pluginType: string,
+  pluginType: EditorPluginType,
   allPluginStrings: EditorStrings['plugins']
 ): SuggestionOption {
-  const pluginData = editorPlugins
+  const pluginData: PluginWithData | undefined = editorPlugins
     .getAllWithData()
     .find((plugin) => plugin.type === pluginType)
 
-  if (!pluginData) return { pluginType, title: pluginType }
+  if (!pluginData) {
+    return { pluginType, title: pluginType }
+  }
 
-  const pluginStrings = Object.hasOwn(allPluginStrings, pluginType)
-    ? allPluginStrings[pluginType as keyof typeof allPluginStrings]
-    : undefined
+  const pluginStrings: PluginStrings | undefined = allPluginStrings[
+    pluginType as keyof typeof allPluginStrings
+  ] as PluginStrings
 
-  const title =
-    pluginStrings && Object.hasOwn(pluginStrings, 'title')
-      ? pluginStrings.title
-      : pluginData.plugin.defaultTitle ?? pluginType
-  const description =
-    pluginStrings && Object.hasOwn(pluginStrings, 'description')
-      ? pluginStrings.description
-      : pluginData.plugin.defaultDescription
-  const { icon } = pluginData
+  const title: string =
+    pluginStrings?.title ?? pluginData.plugin.defaultTitle ?? pluginType
+  const description: string | undefined =
+    pluginStrings?.description ?? pluginData.plugin.defaultDescription
+  const icon: JSX.Element | undefined = pluginData.icon
```

Comment:
why do you initialize the variables with types in this file?

Reply:
Removed